### PR TITLE
Implement `MVP` pattern

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

The ESLint result:
image

File: app/page/login-page/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    this._initialRender();
+  }
+
+  /**
+   * Return entered login data.
+   *
+   * @return {LoginData} - a login data.
+   */
+  get loginData() {
+    return this._loginForm.loginData;
```
Comment (dkalinin) [inline app/page/login-page/index.js]: Let's pass all login data in callback.

File: app/page/registration-page/index.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+   * Performs initial rendering of the component and creates subcomponents. May be overridden,
+   * but this will require calling the parent method: <code> super._initialRender(); </code>.
+   *
+   * @private
+   */
+  _initialRender() {
+    new Component(this.element, 'h1').element.textContent = 'Registration page';
+    this._registrationFormComponent = new RegistrationForm(this.element, {
+      id: 'registration-form',
+    }).onFormSubmit(() => this._registrationSubmittedCallback());
```
Comment (dkalinin) [inline app/page/registration-page/index.js]: Let's pass all data in callback. So there's no need to have getters.

File: app/presenter/login-presenter/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  // TODO:2019-08-22:Oleh.Stefanovych: Method uses demo facade to emulate backend response.
+  // A RequestService should be used when it is implemented.
+  // TODO:2019-08-22:Oleh.Stefanovych: Implement redirection on successful login.
+  /**
+   * A handler that will be executed when the login form is submitted.
+   * @private
+   */
+  _loginningSubmittedHandler() {
```
Comment (dkalinin) [inline app/presenter/login-presenter/index.js]: There's no word "loginning".

File: app/presenter/login-presenter/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    if (this._isHandlerBusy) {
+      return;
+    }
+    this._isHandlerBusy = true;
+
+    const loginData = this.page.loginData;
```
Comment (dkalinin) [inline app/presenter/login-presenter/index.js]: Let's have loginData as parameter in the handler.

File: app/service/ig-generation-service/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/**
+ * A class which that provides a method for generating unique string identifiers for assigning to
+ * DOM elements.
+ */
+export class IdGenerationService {
```
Comment (dkalinin) [inline app/service/ig-generation-service/index.js]: There's a typo in file name: ''i-g - generatiion..."

File: app/service/request-service/index.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+   * Sends the POST HTTP request to the server.
+   *
+   * @param {string} url - an URL to which the request will be sent.
+   * @param {string} json - a body of the request to be sent.
+   * @return {Promise<any>} - a server response.
+   */
+  sendPost(url, json) {
+    return new Promise(function(resolve, reject) {
+      console.log(`Class: RequestService, Function: sendPost, Line 8. I'm trying to send POST on 
+      URL: ${url} with body: ${json}, but I'm not implemented yet.`);
```
Comment (dkalinin) [inline app/service/request-service/index.js]: Let's resolve/reject it in some timeout.

File: app/service/request-service/login-request-service-facade/index.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+// removed.
+/**
+ * @inheritDoc
+ */
+export class LoginRequestServiceFacade extends RequestService {
+  /**
+   * @override
+   * @inheritDoc
+   */
+  sendPost(url, json) {
```
Comment (dkalinin) [inline app/service/request-service/login-request-service-facade/index.js]: `RequestService` is responsible for request sending. If there's a service responsible for logging in then it should have corresponding methods like `logIn(credentials)`. The same for registration service.

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

**Comment by dkalinin** [comment]
> _No description provided._

Don't forget about PR description

**Pull request merged**